tests/test_rm/integration/test_rm_int.py

```diff
@@ -0,0 +1,178 @@
+import os
+import glob
+import shutil
+import radical.pilot as rp
+from radical.pilot.agent import rm as rpa_rm
+from radical.pilot.agent.rm.pbspro import PBSPro
+
+try:
+    import mock
+except ImportError:
+    from unittest import mock
+
+# ------------------------------------------------------------------------------
+#
+def setUp(resource):
+
+    session      = rp.Session()
+    cfg          = session.get_resource_config(resource=resource)
+
+    return cfg, session
+
+
+# ------------------------------------------------------------------------------
+#
+def tearDown():
+
+    #cur_dir       = os.path.dirname(os.path.abspath(__file__))
+    cur_dir = os.getcwd()
+    rp = glob.glob('%s/rp.session.*' % cur_dir)
+
+    for fold in rp:
+        shutil.rmtree(fold)
+
+    try:
+        os.unlink('radical.saga.api.log')
+        os.unlink('radical.saga.log')
+        os.unlink('radical.utils.log')
+    except:
+        pass
+
+
+# ------------------------------------------------------------------------------
+#
+def test_rm_fork():
+
+    cfg, session = setUp('local.localhost')
+    cfg['cores'] = 1
+    cfg['gpus'] = 0
+
+    lrms = rpa_rm.RM.create(name=cfg['lrms'], cfg=cfg, session=session)
+
+    assert lrms.lrms_info == {'agent_nodes'     : {},
+                              'gpus_per_node'   : 1,
+                              'cores_per_node'  : 8,
+                              'lfs_per_node'    : {'path': "/tmp", 'size': 1024},
+                              'lm_info'         : {'version_info'    : {'FORK':
+                                  {'version': '0.42', 'version_detail': 
+                                  'There is no spoon'}}},
+                              'mem_per_node'    : 0,
+                              'name'            : 'Fork',
+                              'node_list'       : [['localhost', 'localhost_0']]}
+    lrms.stop()
+    session.close()
+    tearDown()
+
+
+@mock.patch.object(PBSPro, '_parse_pbspro_vnodes', return_value=['nodes1', 'nodes2'])
```

Comment:
I do not think `_parse_pbspro_vnodes` should be mocked. I suggest that `subprocess.check_output` gets mocked as done [here](https://github.com/radical-cybertools/radical.pilot/blob/devel/tests/test_rm/test_unit/test_pbspro.py#L126) and let PBSPro to run fully.

Reply:
This being an *integration* test, shouldn't those values be actually read from the resource instead of being mocked?